chemicaljson.py

```diff
@@ -0,0 +1,256 @@
+
+
+class Spectra(BaseModel):
+    """Spectra (optional)
+
+    Objects for non-vibrational spectra, including electronic, NMR, and other spectra.
+    """
+
+    electronic: Optional[Electronic] = Field(None, description="Optional electronic spectra")
+    nmr: Optional[Nmr] = Field(None, description="Optional NMR spectra")
+
+
+class Properties(BaseModel):
+    """Properties of the molecule / system. (Optional)
+
+    A set of key-value properties.
+    """
+
+    molecularMass: Optional[float] = None
+    meltingPoint: Optional[float] = None
+    boilingPoint: Optional[float] = None
+    totalCharge: Optional[int] = Field(0, description="Total charge of the system. If omitted, assume 0 (charge neutral)")
+    spinMultiplicity: Optional[int] = Field(1, description="Total spin multiplicity of the system (2S+1, e.g., 1, 2, 3, etc.). If omitted, assume to be 1 (singlet)")
+    totalEnergy: Optional[float] = Field(None, description="Optional total energy of the system in eV")
+
+
+class InputParameters(BaseModel):
+    """Input parameters for the calculation. (Optional)
+
+    Attributes:
+        basis: Basis set used for the calculation (e.g. "6-31G(d)" or "Custom").
+        theory: Method used for the calculation (e.g. "DFT" or "HF" or "MP2").
+        functional: Functional used for the calculation if DFT (e.g. "B3LYP" or "Custom").
+        task: "Energy" or "Optimize" or "Frequencies" or "Transition State" or "Custom".
+    """
+
+    basis: Optional[str] = None
+    functional: Optional[str] = None
+    task: Optional[str] = None
+    theory: Optional[str] = None
+
+
+class PartialCharges(BaseModel):
+    """Partial charges for the atoms in the molecule. (Optional)
+
+    Keys represent the partial charge method, followed by the computed partial charges.
+    e.g.
+    - "Mulliken": [ 0.01, 0.02, 0.03, ... ]
+    - "Gasteiger": [ 0.01, 0.02, 0.03, ... ]
+    """
+
+    Mulliken: List[float]
+
+
+class UnitCell(BaseModel):
+    """Unit cell for the system. (Optional)
+
+    Current versions of Avogadro will output and preferentially use cellVectors,
+    since they fully specify the unit cell, but will also output a, b, c,
+    alpha, beta, gamma parameters and use them if no cellVectors field is found.
+    """
+
+    a: float = Field(..., description="Unit cell a-axis length (in Angstrom).")
+    b: float = Field(..., description="Unit cell b-axis length (in Angstrom).")
+    c: float = Field(..., description="Unit cell c-axis length (in Angstrom).")
+    alpha: float = Field(..., description="Unit cell alpha angle (in degrees).")
+    beta: float = Field(..., description="Unit cell beta angle (in degrees).")
+    gamma: float = Field(..., description="Unit cell gamma angle (in degrees).")
+    cellVectors: Optional[List[float]] = Field(
+        min_items=9, max_items=9, default_factory=0.0, description="Optional list of cell vectors (in Angstrom): [ x1, y1, z1, x2, y2, z2, ... ]"
```

Comment:
This has to be a callable, like
```suggestion
        min_items=9, max_items=9, default_factory=lambda: [0.0 for _ in range(9)], description="Optional list of cell vectors (in Angstrom): [ x1, y1, z1, x2, y2, z2, ... ]"
```